### PR TITLE
[Monorepo] import gutenberg-mobile as a package

### PR DESCRIPTION
**Pull request by dratwas**
## Description

Explores #11491.
Superseeds #17456 started by @Tug 

Wordpress-iOS PR - https://github.com/wordpress-mobile/WordPress-iOS/pull/12762
Wordpress-Android PR - https://github.com/wordpress-mobile/WordPress-Android/pull/10668

This is a migration of gutenberg-mobile to gutenberg repository.

New packages added:

- `@wordpress/react-native-editor` - playground react-native app and bundle entrypoint
- `@wordpress/react-native-bridge` - module which starts react-native inside the client app (`WP-iOS/Android`)
- `@wordpress/react-native-aztec` - react-native-aztec module

Usual `yarn` commands ran from gutenberg-mobile are now accessible from gutenberg using `npm run native`

For instance:

`yarn start` => `npm run native start`

## How has this been tested?
TBD

## Types of changes
Add new native packages, for native build, unifying the 2 repositories

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [x] I've imported git history
- [ ] I've checked that the licencing is OK
- [ ]  I've checked that package release is working as expected (lerna)
- [x]  I've checked that installing dependencies works as expected and that changes to package-lock.json are OK
- [ ] I've checked that we are up to date with gutenberg-mobile develop branch
- [ ] I've informed concerned developers of the change of repository
- [x] I've ported e2e tests iOS
- [x] I've ported e2e tests Android
- [x] I've ported pre-commit hooks including eslint
- [x] I've ported i18n
- [x] I've checked that build scripts are up to date with WordPress-iOS
- [x] I've checked that build scripts are up to date with WordPress-Android
- [x] I've checked that WordPress-iOS works properly after migration
- [x] I've checked that WordPress-Android works properly after migration
- [x] I've made sure that the project can be opened in Android Studio
- [x] I've made sure that the project can be opened in XCode
- [x] I've made sure that the demo app for mobile runs
- [ ] I've made sure that I have excluded any binary files, jar, phar... from the patch (gutenberg-mobile has gradle-wrapper.jar committed to the repo though and we might want to fix that later?)
- [ ] I've checked that all CI steps are imported
- [ ] i've checked that the `react-native-aztec` example app works

### CODE REVIEW

**Comment by gziolo** [comment]
For reference, an old PR where @omarreiss moved https://github.com/WordPress/packages with all git history to Gutenberg: https://github.com/WordPress/gutenberg/pull/7556. I'm sure we should take the same approach here to respect all contributors.

The magic trick happened with:
`git subtree add --prefix temp-packages https://github.com/WordPress/packages.git master`

File: webpack.config.js [hunk 1]
```diff
@@ -29,6 +29,11 @@ const WORDPRESS_NAMESPACE = '@wordpress/';
 
 const gutenbergPackages = Object.keys( dependencies )
 	.filter( ( packageName ) => packageName.startsWith( WORDPRESS_NAMESPACE ) )
+	// TODO fix config to build them as well.
+	.filter( ( packageName ) =>
+		! ( packageName.includes( 'react-native-editor' ) ) &&
```
Comment (Tug) [inline webpack.config.js]: Could simply use `packageName.startsWith( 'react-native' )`.
Btw do you think we can have webpack build our native bundles? That would be awesome :)

Comment (dratwas) [reply webpack.config.js]: >Could simply use packageName.startsWith( 'react-native' ).

Yeah, good idea :)

>Btw do you think we can have webpack build our native bundles? That would be awesome :)

This is what Haul does, so it is possible, but TBH i'm not sure if it should be a part of this PR :) WDYT?

Comment (Tug) [reply webpack.config.js]: > This is what Haul does, so it is possible

Awesome :)

> but TBH i'm not sure if it should be a part of this PR :) WDYT?

Yeah probably not.
If we could have minimal changes in this PR that would be great

**Comment by dratwas** [comment]
MOVED TO https://github.com/WordPress/gutenberg/pull/18159 since we can not use a fork with Jitpack build.

**Pull request closed without merging**